cloudflare.mjs

```diff
@@ -30,9 +32,14 @@ export default async function getDeploymentUrl(
   core.info(`Found ${data.result.length} deployments`)
   core.debug(`Looking for matching deployments ${repo}/${branch}`)
   const builds = data.result
-    .filter((d) => d.environment === 'preview')
     .filter((d) => d.source.config.repo_name === repo)
     .filter((d) => d.deployment_trigger.metadata.branch === branch)
+    .filter((d) => environment == null || d.environment === environment)
+    .filter(
+      (d) =>
+        commitHash == null ||
+        d.deployment_trigger.metadata?.commit_hash === commitHash
```

Comment:
JavaScript Actions still run on Node 12, the optional chaining doesn't work there. You can try to change the babel/ncc config to covert this, or refactor.